tests/integration/pm_tests.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+use deno_core::serde_json::json;
+use test_util::assert_contains;
+use test_util::env_vars_for_jsr_tests;
+// use test_util::env_vars_for_npm_tests;
+// use test_util::itest;
+use test_util::TestContextBuilder;
+
+#[test]
+fn add_basic() {
+  let starting_deno_json = json!({
+    "name": "@foo/bar",
+    "version": "1.0.0",
+    "exports": "./mod.ts",
+  });
+  let context = pm_context_builder().build();
+  let temp_dir = context.temp_dir().path();
+  temp_dir.join("deno.json").write_json(&starting_deno_json);
+
+  let output = context.new_command().args("add @denotest/add").run();
+  output.assert_exit_code(0);
+  let output = output.combined_output();
+  assert_contains!(output, "Add @denotest/add");
+  let deno_json_content = temp_dir.join("deno.json").read_json_value();
+  let expected_deno_json = json!({
+    "name": "@foo/bar",
+    "version": "1.0.0",
+    "exports": "./mod.ts",
+    "imports": {
+      "@denotest/add": "jsr:@denotest/add@^1.0.0"
+    }
+  });
+  assert_eq!(deno_json_content, expected_deno_json);
```

Comment:
This could be:

```rust
temp_dir.join("deno.json").assert_matches_json(json!({
    "name": "@foo/bar",
    "version": "1.0.0",
    "exports": "./mod.ts",
    "imports": {
      "@denotest/add": "jsr:@denotest/add@^1.0.0"
    }
  }));
```